plugins/catalog-backend-module-ldap/src/processors/LdapOrgEntityProvider.ts

```diff
@@ -173,6 +205,38 @@ export class LdapOrgEntityProvider implements EntityProvider {
 
     markCommitComplete();
   }
+
+  private schedule(schedule: LdapOrgEntityProviderOptions['schedule']) {
+    if (schedule === 'manual') {
+      return;
+    }
+
+    this.scheduleFn = async () => {
+      const id = this.getScheduledTaskId();
+      await schedule.run({
+        id,
+        fn: async () => {
+          const logger = this.options.logger.child({
+            class: LdapOrgEntityProvider.prototype.constructor.name,
+            taskId: id,
+            taskInstanceId: uuid.v4(),
+          });
+
+          try {
+            await this.read({ logger });
+          } catch (error) {
+            logger.error(error);
+          }
+        },
+      });
+    };
+  }
+
+  // Gets a suitable scheduler task ID for this provider instance
+  private getScheduledTaskId(): string {
+    const rawId = `refresh_${this.getProviderName()}`;
+    return rawId.toLocaleLowerCase('en-US').replace(/[^a-z0-9]/g, '_');
```

Comment:
Do we really need this transform? Is it not a bit more straight-forward to use the provider name as is?

Reply:
The tasks package enforces a schema on task ids, to make them compatible with prometheus which is picky about its IDs. Could be argued that THAT should be hidden inside the tasks package, but it isn't atm

Reply:
Prometheus label values allow any unicode characters though, which is where you'd have these IDs end up?

https://prometheus.io/docs/concepts/data_model/#metric-names-and-labels

Reply:
Hm! Interesting. Well in any case, the tasks package DOES enforce this currently for better or worse, so I'd have to investigate and fix that separately.

Reply:
relaxed!